### DISCUSSION TITLE
How to properly hydrate Redux state in NextJS on page refresh?

### DISCUSSION THREAD

**Discussion by bhatvikrant** [vercel/next.js#21238] [Help] [answered]
The problem I'm having is hydrating the user data from local storage on app reload/page refresh.

In my project, I am using NextJS for frontend, and for support libraries, I am using redux-toolkit for redux management across the application.

The user can log in, and in that case, I store the isLoggedIn boolean in local storage and in the redux state. Depending on the isLoggedIn boolean value I change the Navbar component styles (the Navbar is included directly in _app.tsx).

When the user refreshes any page the isLoggedIn boolean is not loaded into the redux state but is present in local storage.

How can I update my Redux state too whenever a full page refresh is done? 

> PS: I feel that I might need to put a useEffect somewhere which updates my redux store, In a Create react app project this could have been done in the App.js file but where can a similar thing be done in NextJS? 
> I am also aware of the App.getInitialProps in _app.tsx but I also don't wanna loose the benefits of static page optimisation.

**Comment by enochndika** [comment, marked as answer]
You can include it on your _app.js or in your navbar file

now your state will be persisted when a full refresh is done
```
  const isLoggedIn =
    typeof window !== "undefined" ? localStorage.getItem("isLoggedIn") : null;
  const dispatch = useDispatch();

  useEffect(() => {
    if (isLoggedIn) {
      dispatch(/* your action*/);
    }
  }, [isLoggedIn]);
```
